amazon_gem.sh

```diff
@@ -46,7 +50,23 @@ for _package in rubygems ruby-devel augeas augeas-devel augeas-libs; do
   _install_package "$_package" || rc=$?
 done
 for _gem in puppet ruby-augeas ruby-nagios; do
-  _install_gem "$_gem" || rc=$?
+
+  # Scan through commands to see if specific versions are set
+  while [ $# -gt 0 ]
```

Comment:
Would move this out of for loop; options handling should happen first.
Also another way to do this with a case statement:
```bash
while [[ -n "$1" ]]; do
  case "$1" in
    --puppetver) shift
      _puppetver="$1"
      ;;
  esac
  shift
done
```